src/BenchmarkDotNet/Toolchains/GeneratorBase.cs

```diff
@@ -119,7 +121,18 @@ private ArtifactsPaths GetArtifactsPaths(BuildPartition buildPartition, string r
             string programName = buildPartition.ProgramName;
             string buildArtifactsDirectoryPath = GetBuildArtifactsDirectoryPath(buildPartition, programName);
             string binariesDirectoryPath = GetBinariesDirectoryPath(buildArtifactsDirectoryPath, buildPartition.BuildConfiguration);
-            string executablePath = Path.Combine(binariesDirectoryPath, $"{programName}{GetExecutableExtension()}");
+
+            string executablePath;
+            if (buildPartition.IsWasm)
+            {
+                executablePath = Path.Combine(binariesDirectoryPath, "run-v8.sh");
+            }
+            else
+            {
+                executablePath = Path.Combine(binariesDirectoryPath, $"{programName}{GetExecutableExtension()}");
+            }
+
+            executablePath = GetExecutablePath(binariesDirectoryPath, programName);
```

Comment:
the whole block above isn't necessary now that you have `GetExecutablePath`:

```suggestion
            string executablePath = GetExecutablePath(binariesDirectoryPath, programName);
```